c7n/resources/eks.py

```diff
@@ -146,7 +146,54 @@ class Delete(Action):
     def process(self, resources):
         client = local_session(self.manager.session_factory).client('eks')
         for r in resources:
+            nodegroups = client.list_nodegroups(clusterName=r['name'])['nodegroups']
+            fargateProfileNames = client.list_fargate_profiles(
+                clusterName=r['name'])['fargateProfileNames']
+            if nodegroups:
+                for nodegroup in nodegroups:
+                    client.delete_nodegroup(
+                        clusterName=r['name'], nodegroupName=nodegroup)
+                    waiter = client.get_waiter('nodegroup_deleted')
```

Comment:
issue all the deletes before the waits (with self.manager.retry), before waiting

Reply:
This works like a charm in parallel 👌 updated pr on the way

Reply:
Added in 96fb414